### PR TITLE
fix index check for bytevector-reference functions

### PR DESCRIPTION
**Pull request by mflatt**
Intended as a repair for #821.

### CODE REVIEW

**Review by burgerrg — APPROVED** [review]

File: s/prims.ss [hunk 1]
```diff
@@ -88,7 +88,8 @@
   (lambda (bv i)
     (unless (reference-bytevector? bv) ($oops who "~s is not a reference bytevector" bv))
     (unless (and (fixnum? i)
-                 (not ($fxu< (fx- (bytevector-length bv) (fx- (constant ptr-bytes) 1)) i)))
+                 (fx>= i 0)
+                 (fx<= i (fx- (bytevector-length bv) (constant ptr-bytes))))
```
[the remark this answers is not in the document]
Comment (mflatt) [reply s/prims.ss]: Yes, that seems better. Changed!

**Comment by mflatt** [comment]
Looks like I need to adjust tests so that error tests produce output independent of the word size.

**Comment by burgerrg** [comment]
Using any positive size less than 4 should do the trick.

**Review by burgerrg — APPROVED** [review]
Thank you for fixing this!

**Pull request merged**